### PR TITLE
Fixes some disposals bugs

### PR DESCRIPTION
**Pull request by NoooneyDude**
### Purpose
- Players can now exit disposal bins by attempting to move, alleviating issue #4646. Fixes part 12. of issue #4654.
- Clicking the transit tubes toggle in the pipe dispenser no longer causes a server error. Partially fixes part 10. of #4654.
- Attempting to move while in a disposals pipe will now produce a clang sound. Fixes part D. of #4654.
- Fixes a few compiler warnings. Fixes part 1. of #4654.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Disposals/DisposalBin.cs [hunk 1]
```diff
@@ -22,7 +22,10 @@ public class DisposalBin : DisposalMachine, IServerDespawn, IExaminable, IChecke
 		const int AUTO_FLUSH_DELAY = 2;
 		const float ANIMATION_TIME = 1.3f; // As per sprite sheet JSON file.
 
+// Ignore never-assigned-to warning as it is assigned via inspector
+#pragma warning disable CS0649
 		[SerializeField] AudioSource rechargeSFX;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Disposals/DisposalBin.cs]: if you do [SerializeField] AudioSource rechargeSFX = null; should get rid of warning

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Disposals/DisposalBin.cs]: Is that preferred over warning disables?

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Disposals/DisposalBin.cs]: I think so yes, i havnt seen warning disables any where else.

**Comment by ThatDan123** [comment]
Does it actually fix #4646 though? As the teleporting would still be annoying.

File: UnityProject/Assets/Scripts/Disposals/DisposalPipe.cs [hunk 2]
```diff
@@ -7,6 +7,8 @@ namespace Disposals
 	[CreateAssetMenu(fileName = "DisposalPipe", menuName = "Tiles/Disposal Pipe", order = 1)]
 	public class DisposalPipe : BasicTile, IExaminable
 	{
+// Ignore never-assigned-to warning as it is assigned via inspector
+#pragma warning disable CS0649
 		[Tooltip("Set the type of disposal pipe this is.")]
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Disposals/DisposalPipe.cs]: Dont need to do the warning things here either. For lists do ```= new List<type>();```

**Comment by NoooneyDude** [comment]
> Does it actually fix #4646 though? As the teleporting would still be annoying.

No. Will take a look at the next opportunity.

File: UnityProject/Assets/Scripts/Disposals/DisposalPipeObject.cs [hunk 3]
```diff
@@ -13,6 +13,8 @@ public class DisposalPipeObject : NetworkBehaviour, IExaminable, ICheckedInterac
 		TileChangeManager tileChangeManager;
 		ObjectBehaviour behaviour;
 
+// Ignore never-assigned-to warning as it is assigned via inspector
+#pragma warning disable CS0649
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Disposals/DisposalPipeObject.cs]: And the warnings here.

**Pull request merged**